src/openms/include/OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.h

```diff
@@ -95,56 +89,26 @@ namespace OpenMS
       DefaultParamHandler::operator=(source);
       ProgressLogger::operator=(source);
       stn_estimates_ = source.stn_estimates_;
-      first_ = source.first_;
-      last_  = source.last_;
       return *this;
     }
 
     /// Destructor
     ~SignalToNoiseEstimator() override
     {}
 
-
     /// Set the start and endpoint of the raw data interval, for which signal to noise ratios will be estimated immediately
-    virtual void init(const PeakIterator & it_begin, const PeakIterator & it_end)
-    {
-      first_ = it_begin;
-      last_ = it_end;
-      computeSTN_(first_, last_);
-      is_result_valid_ = true;
-    }
-
-    /// Set the start and endpoint of the raw data interval, for which signal to noise ratios will be estimated immediately
-    virtual void init(const Container & c)
-    {
-      init(c.begin(), c.end());
-    }
-
-    /// Return to signal/noise estimate for data point @p data_point
-    /// @note the first query to this function will take longer, as
-    ///       all SignalToNoise values are calculated
-    /// @note you will get a warning to stderr if more than 20% of the
-    ///       noise estimates used sparse windows
-    virtual double getSignalToNoise(const PeakIterator & data_point)
+    virtual void init(const Container& c)
     {
-      if (!is_result_valid_)
-      {
-        // recompute ...
-        init(first_, last_);
-      }
-
-      return stn_estimates_[*data_point];
+      computeSTN_(c);
     }
 
-    virtual double getSignalToNoise(const PeakType & data_point)
+    ///Return to signal/noise estimate for date point @p index
+    ///@note you will get a warning to stderr if more than 20% of the
+    ///      noise estimates used sparse windows
+    virtual double getSignalToNoise(const Size index)
```

Comment:
could be `virtual double getSignalToNoise(const Size index) const` now?

Reply:
why could be this function `const`?

Reply:
because it does read-only access to the underlying member, i.e. does not change the state of the object